File: src/main/java/com/github/seratch/jslack/app_backend/events/payload/EventsApiPayload.java
package com.github.seratch.jslack.app_backend.events.payload;

import com.github.seratch.jslack.api.model.event.Event;

/**
 * https://api.slack.com/types/event
 */
public interface EventsApiPayload<E extends Event> {

    String getToken();

    void setToken(String token);

    String getTeamId();

    void setTeamId(String teamId);

    String getApiAppId();

    void setApiAppId(String apiAppId);

    E getEvent();

    void setEvent(E event);

    String getType();

    void setType(String type);

    String getEventId();

    void setEventId(String eventId);

    Integer getEventTime();

    void setEventTime(Integer eventTime);

}

File: typescript-types/events-api/express-support/EventsApiPayload.ts
export interface EventsApiPayload {
  token?: string;
  team_id?: string;
  api_app_id?: string;
  event?: Event;
  type?: string;
  authed_users?: string[];
  event_id?: string;
  event_time?: number;
}

export interface Event {
  type?: string;
  [key: string]: any;
}